src/server.lisp

```diff
@@ -0,0 +1,185 @@
+                     (unknown-rpc-method-name condition)))))
+
+
+(defun %worker-routine (&key
+                          (dispatch-table (error "bad value"))
+                          (logging-stream (error "bad value"))
+                          timeout)
+  "The thread body for an RPCQ server.  Responds to RPCQ requests which match entries in DISPATCH-TABLE."
+  (pzmq:with-socket receiver :dealer
+    (pzmq:connect receiver "inproc://workers")
+    (loop
+      (handler-case
+          (let (request result reply)
+            (multiple-value-bind (identity empty-frame raw-request)
+                (%pull-raw-request receiver)
+              
+              (handler-case
+                  (progn
+                    (setf request (deserialize raw-request))
+                    (unless (typep request '|RPCRequest|)
+                      (error 'not-an-rpcrequest
+                             :object request))
+                    (format-log logging-stream "Got request: ~a" (|RPCRequest-method| request))
+                    
+                    (let ((params-as-plist
+                            (loop :for key :being :the :hash-keys :of (|RPCRequest-params| request)
+                                  :with val := (gethash key (|RPCRequest-params| request))
+                                  :collect (str->lisp-keyword key)
+                                  :collect val)))
+                      (let ((f (gethash (|RPCRequest-method| request) dispatch-table)))
+                        (unless f
+                          (error 'unknown-rpc-method
+                                 :method-name (|RPCRequest-method| request)))
+                        (setf result 
+                              (if timeout
+                                  (bt:with-timeout (timeout)
+                                    (apply f params-as-plist))
+                                  (apply f params-as-plist)))
+                        
+                        (setf reply (make-instance '|RPCReply|
+                                                   :|id| (|RPCRequest-id| request)
+                                                   :|result| result))))
+                    (format-log logging-stream "Finishing request: ~a" (|RPCRequest-method| request)))
+                
+                ;; this is where errors go where we can reply to the client
+                (unknown-rpc-method (c)
+                  (declare (ignore c))
+                  (format-log logging-stream "Error: method ~a unknown" (|RPCRequest-method| request))
+                  (setf reply (make-instance '|RPCError|
+                                             :|id| (|RPCRequest-id| request)
+                                             :|error| (format nil "Method named \"~a\" is unknown."
+                                                              (|RPCRequest-method| request)))))
+                (sb-ext:timeout (c)
```

Comment:
```suggestion
                (bt:timeout (c)
```

Reply:
! I didn't know that it also offered an exception type alias. Great!